sky/spot/spot_utils.py

```diff
@@ -564,7 +582,7 @@ def is_spot_controller_up(
     """
     try:
         controller_status, handle = backend_utils.refresh_cluster_status_handle(
-            SPOT_CONTROLLER_NAME, force_refresh=True)
+            SPOT_CONTROLLER_NAME)
```

Comment:
Sorry for being back and forth, it seems we should set the `force_refresh=False`, as the optimization is critical in reducing the time for fetching spot jobs from 38s to 9s, when the controller is STOPPED.

Added a comment as follows:
```
# Set force_refresh=False to make sure the refresh only happens when the
# controller is INIT/UP. This optimization avoids unnecessary
# costly refresh when the controller is already stopped. This
# optimization is based on the assumption that the user will not start
# the controller manually from the cloud console.
```

It is OK to set this option to `False`, because keeping `force_refresh=True` only has limited ability to "fix the INIT status caused by the user Ctrl-C" as mentioned above.

PTAL. : )

Reply:
Great catch!